src/MerchantCenter/AccountService.php

```diff
@@ -0,0 +1,491 @@
+	 *
+	 * @param int $merchant_id The Merchant Center account to update.
+	 *
+	 * @throws ExceptionWithResponseData If the account URL doesn't match the site URL or the URL is invalid.
+	 */
+	private function maybe_add_merchant_center_url( int $merchant_id ) {
+		$site_url = esc_url_raw( $this->get_site_url() );
+
+		if ( ! wc_is_valid_url( $site_url ) ) {
+			throw $this->prepare_exception( __( 'Invalid site URL.', 'google-listings-and-ads' ) );
+		}
+
+		/** @var Merchant $merchant */
+		$merchant = $this->container->get( Merchant::class );
+
+		/** @var Account $account */
+		$account     = $merchant->get_account( $merchant_id );
+		$account_url = $account->getWebsiteUrl();
+
+		if ( untrailingslashit( $site_url ) !== untrailingslashit( $account_url ) ) {
+
+			$is_website_claimed = $merchant->get_accountstatus( $merchant_id )->getWebsiteClaimed();
+
+			if ( ! empty( $account_url ) && $is_website_claimed && ! $this->allow_switch_url ) {
+				$state                              = $this->state->get();
+				$state['set_id']['data']['old_url'] = $account_url;
+				$state['set_id']['status']          = MerchantAccountState::STEP_ERROR;
+				$this->state->update( $state );
+
+				$clean_account_url = $this->strip_url_protocol( $account_url );
+				$clean_site_url    = $this->strip_url_protocol( $site_url );
+
+				do_action( 'woocommerce_gla_url_switch_required', [] );
+
+				throw $this->prepare_exception(
+					sprintf(
+					/* translators: 1: is a website URL (without the protocol) */
+						__( 'This Merchant Center account already has a verified and claimed URL, %1$s', 'google-listings-and-ads' ),
+						$clean_account_url
+					),
+					[
+						'id'          => $merchant_id,
+						'claimed_url' => $clean_account_url,
+						'new_url'     => $clean_site_url,
+					],
+					409
+				);
+			}
+
+			$account->setWebsiteUrl( $site_url );
+			$merchant->update_account( $account );
+
+			do_action( 'woocommerce_gla_url_switch_success', [] );
+		}
+	}
+
+	/**
+	 * Prepares an Exception to be thrown with Merchant data:
+	 * - Ensure it has the merchant_id value
+	 * - Default to a 400 error code
+	 *
+	 * @param string   $message
+	 * @param array    $data
+	 * @param int|null $code
+	 *
+	 * @return ExceptionWithResponseData
+	 */
+	private function prepare_exception( string $message, array $data = [], int $code = null ): ExceptionWithResponseData {
+		$merchant_id = $this->options->get_merchant_id();
+
+		if ( $merchant_id && ! isset( $data['id'] ) ) {
+			$data['id'] = $merchant_id;
+		}
+
+		return new ExceptionWithResponseData( $message, $code ?: 400, null, $data );
+	}
+
+	/**
+	 * Removes the protocol (http:// or https://) and trailing slash from the provided URL.
+	 *
+	 * @param string $url
+	 *
+	 * @return string
+	 */
+	private function strip_url_protocol( string $url ): string {
+		return preg_replace( '#^https?://#', '', untrailingslashit( $url ) );
+	}
```

Comment:
💅 I think this is a pretty generic function that maybe has better placement in some kind of helper file.

Reply:
I did consider this but thought it might be oddly placed if no other class uses it. I moved it to the PluginHelper trait.